packages/driver/index.ts

```diff
@@ -0,0 +1,48 @@
+import { Operation } from 'effection';
+
+export interface DriverSpec<TOptions = unknown> {
+  module: string;
+  options?: TOptions;
+}
+
+export interface Driver<TData = unknown> {
+  description: string;
+  data: TData;
+  connect(agentURL: string): Operation<void>;
+}
+
+export interface DriverFactory<TOptions, TData> {
+  (spec: DriverSpec<TOptions>): Operation<Driver<TData>>;
+}
+
+export function load<TOptions, TData>(spec: DriverSpec<TOptions>): Operation<Driver<TData>> {
+  try {
+    let exports = require(spec.module);
```

Comment:
Should this use `import` rather than `require`? I'm a bit unsure of which is preferred?

Reply:
The primary difference is that `import` is async, whereas `require` is synchronous which would mean that we'd have to make `load` itself an async operation. This would definitely be more flexible from a future-proof standpoint in the event that we wanted to do some stuff like lazily initialize plugins or something.